Automatise numerical FONLL (#153)
NNPDF/pineko: docs/source/overview/running.rst

If necessary it is possible to specify the values of the *renormalization* and *factorization* scale variations with
the options ``--xir`` and ``--xif``.

Generating FONLL FK Tables
--------------------------

In order to generate FK tables with (numerical) |FONLL| the different flavor schemes
need to be evolved separately and joined together in a single FK table only
in the final step.

There are 2 workflows possible, one in which all the steps are performed individually

> and one that is more automatize.

Reviewer: ```suggestion
and one that is more automatized.
```

The automatized workflows always assumes that |FONLL| is performed both for
charm and bottom effects.

The manual procedure works as follow:

  1. Generate 7 theories for all the different flavor patches with command::

      pineko fonll_tcards THEORY_ID

    The different flavor patches are named following the convention:

>       * ``THEORY_ID00`` : |FFNS| :math:`n_f=3`
>       * ``THEORY_ID01`` : |FFN0| :math:`n_f=3`
>       * ``THEORY_ID02`` : massless component of |FFNS| :math:`n_f=4`
>       * ``THEORY_ID03`` : massive component of |FFNS| :math:`n_f=4`
>       * ``THEORY_ID05`` : massless component of |FFNS| :math:`n_f=5`
>       * ``THEORY_ID06`` : massive component of |FFNS| :math:`n_f=5`

Reviewer 2: ```suggestion
      * ``<THEORY_ID>00`` : |FFNS| :math:`n_f=3`
      * ``<THEORY_ID>01`` : |FFN0| :math:`n_f=3`
      * ``<THEORY_ID>02`` : massless component of |FFNS| :math:`n_f=4`
      * ``<THEORY_ID>03`` : massive component of |FFNS| :math:`n_f=4`
      * ``<THEORY_ID>04`` : |FFN0| :math:`n_f=4`
      * ``<THEORY_ID>05`` : massless component of |FFNS| :math:`n_f=5`
      * ``<THEORY_ID>06`` : massive component of |FFNS| :math:`n_f=5`
```

>     where for |FFNS| :math:`n_f=4,5` massive and massless parts are splitted to

Reviewer 2: ```suggestion
    where for |FFNS| :math:`n_f=4,5` massive and massless parts are split to
```

    allow for a damping option.

>   2. Generate the grids corrisponding to all the 7 theories with the external program.

Reviewer 2: Hmm this is a bit ugly. I understand wanting to keep things modular, but I think we should really have a step-by-step guide somewhere that explains how to use pinefarm/yadism together with pineko to create nFONLL FKtables. We will put it in the paper, but it would also be good to have it in a living document, i.e. docs, somewhere.

Author: True, but this is why I wanted a single place where you can run everything...
As far as `pineko` is concerned we can add a link to `pinefarm` of `pineline` (although the page would be non existing yet.)

  3. Generate the operator cards for each theory with the usual command.
     Note that, in principle, only 3 ekos are needed, as there are only 3 different :math:`n_f` patches.
     So you might speed up the procedure taking advantage of inherit-ekos.

>   4. Generate the ekos for each theory with the usual commands.

Reviewer 2: People who don't know the "usual commands" should also be able to follow this guide. Same goes for the point below

Author: I meant the one listed above. Let me clarify

  5. Generate the FK tables each of the 7 theories with the usual commands.

  6. Combine the various FK tables into a single file, using the command::

      pineko combine_fonll THEORY_ID DATASET --FFNS3 THEORY_ID00 --FFN03 THEORY_ID01 --FFNS4til THEORY_ID02 --FFNS4bar THEORY_ID03 --FFN04 THEORY_ID04 --FFNS5til THEORY_ID05 --FFNS5bar THEORY_ID06

    where the first 3 theories are needed to perform |FONLL| on charm effects,
    while the last 4 are needed to include also bottom effects.

> The automatize procedure can be faster, but it might be less flexible.

Reviewer: ```suggestion
The automatized procedure can be faster, but it might be less flexible.
```

In this case the workflows is:

  1. Generate 7 theories for all the different flavor patches with the command::

      pineko fonll_tcards THEORY_ID

    See above for the intermediate theories naming convention.

>   2. Generate the grids corrisponding to all the 7 theories with the external program.

Reviewer 2: ```suggestion
  2. Generate the grids corresponding to all the 7 theories with the external program.
```

>   3. Generate the ekos for only 3 theories and inherit the others running::

Reviewer 2: ```suggestion
  3. Generate the three ekos, one for each :math:`n_f`, and inherit the others running:
```

      pineko fonll_ekos THEORY_ID DATASET1 DATASET2 ...

    Note: this is usually an expensive operation as multiple ekos are run sequentially.

>     Depending on the resources that you have available it might be more convinient

Reviewer 2: ```suggestion
    Depending on the resources that you have available it might be more convenient
```
